test/common_distributed.py

```diff
@@ -121,39 +123,66 @@ def wrapper(self):
     @classmethod
     def setUpClass(cls):
         for attr in dir(cls):
-            if attr.startswith('test'):
+            if attr.startswith("test"):
                 fn = getattr(cls, attr)
                 setattr(cls, attr, cls.join_or_run(fn))
 
     def setUp(self):
         super(MultiProcessTestCase, self).setUp()
         self.skip_return_code_checks = []
         self.rank = self.MAIN_PROCESS_RANK
-        self.file = tempfile.NamedTemporaryFile(delete=False)
-        self.processes = [self._spawn_process(rank) for rank in range(int(self.world_size))]
+        self.file_name = tempfile.NamedTemporaryFile(delete=False).name
+        self._spawn_process()
```

Comment:
This function now spawns multiple processes and should be renamed accordingly, e.g. `spawn_processes`.